client/app/pages/queries/source-view.js

```diff
@@ -52,7 +52,9 @@ function QuerySourceCtrl(
 
   $scope.canForkQuery = () => currentUser.hasPermission('edit_query') && !$scope.dataSource.view_only;
 
-  $scope.updateQuery = newQueryText => defer(() => $scope.$apply(() => { $scope.query.query = newQueryText; }));
+  $scope.updateQuery = debounce(
+    newQueryText => defer(() => $scope.$apply(() => { $scope.query.query = newQueryText; })),
```

Comment:
Not a blocker, but - when you use `debounce` - `defer` is no longer needed (`debounce` includes its behavior).

Reply:
That makes sense :)